### DISCUSSION TITLE
Show gif in a layer

### DISCUSSION THREAD

**Discussion by LaskySY** [visgl/deck.gl#6463] [Q&A]
Hi guys! I am working on a project that has some special requirements, and I am currently stuck in it. This is a million-level scatter plot, and each point has its own gif, which is a web link. I am required to create a feature to show several points' gif in random each time panning the plot. 
I was trying use iconlayer to fetch gif URL, but it could only show static image which is the first frame of gif. I could not use iconAtlas, because it only shows one gif, rather than based on data.
Also, I tried to use sceneLayer, but we don't have glb or gltf data source.
Our data source: 3D object with obj formate and their gif  
Any ideas?

**Comment by ibgreen** [comment]
We don't support animated textures from GIFs. I believe there is some support for Video textures, e.g MP4 maybe you can poke around https://github.com/visgl/deck.gl/search?q=video

**Comment by Pessimistress** [comment]
You can either pass a video to `iconAtlas`:

```js
new Deck({
  ...
  layers: [
    new IconLayer({
      ...
      iconAtlas: // HTMLVideoElement
    })
  ],
  // Force deck.gl to redraw on every animation frame
  _animate: true
});
```

Or use a sprite image (i.e. each frame of your gif will be a different icon):

[Example](https://codesandbox.io/s/deck-gl-iconlayer-animation-m7ikj)
